python/finufft/_interfaces.py

```diff
@@ -345,12 +345,12 @@ def _rchkf(x):
 def _cchkf(x):
     """
     Check if array x is of the appropriate type
-    (complex128, C-contiguous in memory)
+    (complex64, C-contiguous in memory)
     If not, produce a copy
     """
     if x is not None and (x.dtype is not np.dtype('complex64') and x.dtype is not np.dtype('float32')):
```

Comment:
why are we still allowing float32 here in the *complex* _cchk routine?  I think we should force the user to supply complex.

Reply:
Same applies to the _cchk routine on line 327.  We should only allow complex not real.  Or is there some reason I'm missing?

Reply:
This was the prior behavior. Wasn't really my place to change it. I just put in a patch to ASPIRE to do the cast, so they will be fine if you prefer to raise here instead of handing it for users.

Reply:
I think previous we decided if user given a single precision float32, we will make a new array of single precision complex by "return np.array(x, dtype=np.complex64, order='C')".  We can definitely change to be more strict and require complex input, that is more consistent I guess.

Reply:
Seems we all agrees to remove float32, I guess @garrettwrong  can help to delete `and x.dtype is not np.dtype('float32')` in this PR and we can merge to master.

Reply:
sure I changed the branch to raise on the two real cases and updated the doc string for the util functions.